### PR TITLE
Add EIP-5570: Digital Receipt Non-Fungible Tokens

### PR DESCRIPTION
**Pull request by darcys22**

### CODE REVIEW

File: EIPS/eip-5570.md [hunk 1]
```diff
@@ -0,0 +1,246 @@
+---
+eip: 5570
+title: Digital Receipt Non-Fungible Tokens
+description: Non-Fungible Tokens as a method of distributing digital receipts for physical purchases, where the metadata represents a json receipt
```
Comment (SamWilsn) [inline EIPS/eip-5570.md:4]: Just cleaning up the wording a bit:

```suggestion
description: Non-Fungible Tokens as digital receipts for physical purchases, where the metadata represents a JSON receipt
```

---

I'm not too sure what "the metadata represents a JSON receipt" means. Do you mean something like "with JSON encoded metadata"?

File: EIPS/eip-5570.md [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+created: 2022-09-01
+requires: 721
+---
+
+## Abstract
+
+This EIP proposes a standard schema for digital receipts of transactions. Digital Receipt Non-Fungible Tokens are issued by a vendor when a customer makes a purchase from their store and contains transaction details necessary for record keeping. Digital Receipt Non-Fungible Tokens extend [EIP-721](./eip-721.md) which allows for the management and ownership of unique tokens.
+
+## Motivation
+
```
Comment (SamWilsn) [inline EIPS/eip-5570.md:19]: This section doesn't answer two important questions about this proposal:

 - Why does this information need to exist on-chain?
 - If it does need to exist on-chain, why does it need to be a token (and not some other format, like a registry.)

File: EIPS/eip-5570.md [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+Digital receipt as NFTs can also conceptually include other important information such as item serial numbers and delivery tracking etc.
+
+One of the major roadblocks to fully automating our finance world has been the difficulty in tracking transaction details. Human beings physically tracking paper receipts is archaic and NFTs on the blockchain provide a pathway for these systems to be significantly improved.
+
+## Specification
+
+Transaction Flow:
+ - A customer purchases an item from an online retailer, checking out leads the customer to an option to mint a NFT.
+ - The smart contract provides the user with a Digital Receipt Non-Fungible Token.
+ - When fulfilling the order, the retailer will upload the digital receipt specified in in the JSON schema below as the metadata to the previously minted NFT
```
Comment (SamWilsn) [inline EIPS/eip-5570.md:35]:
```suggestion
 - When fulfilling the order, the retailer will upload the digital receipt specified in in the JSON schema below as the metadata to the previously minted NFT.
```

File: EIPS/eip-5570.md (hunk 1 above)
Comment (darcys22) [reply EIPS/eip-5570.md:4]: This is simply poor English on my part. Intention was to convey that an NFT has metadata, and that metadata can be a json receipt

File: EIPS/eip-5570.md (hunk 2 above)
Comment (darcys22) [reply EIPS/eip-5570.md:19]: Awesome points. Will get some more detail here about those two

**Review by SamWilsn — APPROVED** [review]
This is acceptable for a draft. Looking forward to seeing how this develops!

**Comment by nickjuntilla** [comment]
I recommend adding code to in order to make these receipts accessible by smart contracts. This would be great for other smart contracts to be able to access this data for:

- DeFi
- Royalties
- Decentralized Accounting and Insurance
- Things we probably can't think of right now

I wrote some demo code here:
`
pragma solidity ^0.8.0;
import "./IERC165.sol";

///
/// @dev Interface for eip-5570 Receipt Standard
///
interface IERC5570 is IERC165 {
    /// ERC165 bytes to add to interface array - set in parent contract
    /// implementing this standard
    ///
    /// Below needs to be calculated depending on final interface
    /// bytes4(keccak256("receipt(uint256,uint256)")) == 0xUNSET
    /// bytes4 private constant _INTERFACE_ID_ERC5570 = 0xUNSET;
    /// _registerInterface(_INTERFACE_ID_ERC5570);

    function paid(
        uint256 _tokenId,
    ) external view returns (
        uint256 goodsAndServices,
        uint256 fees,
        uint256 tax,
        bytes8 symbol
    );
    
    function buyer(
        uint256 _tokenId,
    ) external view returns (
        address buyerAddress
    );
    
    function seller(
        uint256 _tokenId,
    ) external view returns (
        address seller
    );
    
    // Optional - Used for the sale of things represented by NFTs
    // Returns all the tokenIds that were sold, their contracts, and prices 
    // The price should be in the symbol designated above.
    // It might be useful to have an NFT reference the receipt it was sold in 
    // and the reference receipt NFT would have to conform to this standard
    function purchasedItems(
        uint256 _tokenId,
    ) external view returns (
        uint256[] tokenId,
        address[] contract,
        uint256[] price
    );
}

interface IERC165 {
    /// @notice Query if a contract implements an interface
    /// @param interfaceID The interface identifier, as specified in ERC-165
    /// @dev Interface identification is specified in ERC-165. This function
    ///  uses less than 30,000 gas.
    /// @return `true` if the contract implements `interfaceID` and
    ///  `interfaceID` is not 0xffffffff, `false` otherwise
    function supportsInterface(bytes4 interfaceID) external view returns (bool);
`
https://github.com/Ownerfy/receipt-standard/blob/main/eip-5570-solidity.sol

File: EIPS/eip-5570.md [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+                  "type": "array",
+                  "items": { "type": "string" },
+                  "minItems": 2,
+                  "maxItems": 6
+                }
+              }
+            }
+          }
+        },
+        "items": {
```
Comment (nickjuntilla) [inline EIPS/eip-5570.md:145]: This should have a bracket for the array of items probably

File: EIPS/eip-5570.md (hunk 2 above)
Comment (Articologist) [reply EIPS/eip-5570.md:19]: It needs to exist for critical mass. It exhibits the greatest adaption through market place purchases. The question is a red-herring.

**Review by Articologist — COMMENTED** [review]
Case Usage: My wife lost her receipt for a cash purchase of a juicer. The store pointed her to the manufacturer. Customer support at the manufacturer laughed and replied, "I dont know why people try to redeem their warranties with no receipt." This is a practical bridge with cash, digital money and warranties.

**Pull request merged**